src/main/java/com/amazon/sqs/javamessaging/ExtendedClientConfiguration.java

```diff
@@ -15,203 +15,138 @@
-	public AmazonS3 getAmazonS3Client() {
-		return s3;
-	}
-
-	/**
-	 * Gets the name of the S3 bucket which is being used for storing
-	 * large-payload messages.
-	 *
-	 * @return The name of the bucket which is being used.
-	 */
-	public String getS3BucketName() {
-		return s3BucketName;
-	}
-
-	/**
-	 * Sets the message size threshold for storing message payloads in Amazon
-	 * S3.
-	 *
-	 * @param messageSizeThreshold
-	 *            Message size threshold to be used for storing in Amazon S3.
-	 *            Default: 256KB.
-	 */
-	public void setMessageSizeThreshold(int messageSizeThreshold) {
-		this.messageSizeThreshold = messageSizeThreshold;
-	}
-
-	/**
-	 * Sets the message size threshold for storing message payloads in Amazon
-	 * S3.
-	 *
-	 * @param messageSizeThreshold
-	 *            Message size threshold to be used for storing in Amazon S3.
-	 *            Default: 256KB.
-	 * @return the updated ExtendedClientConfiguration object.
-	 */
-	public ExtendedClientConfiguration withMessageSizeThreshold(int messageSizeThreshold) {
-		setMessageSizeThreshold(messageSizeThreshold);
-		return this;
-	}
-
-	/**
-	 * Gets the message size threshold for storing message payloads in Amazon
-	 * S3.
-	 *
-	 * @return Message size threshold which is being used for storing in Amazon
-	 *         S3. Default: 256KB.
-	 */
-	public int getMessageSizeThreshold() {
-		return messageSizeThreshold;
-	}
-
-	/**
-	 * Sets whether or not all messages regardless of their payload size should
-	 * be stored in Amazon S3.
-	 *
-	 * @param alwaysThroughS3
-	 *            Whether or not all messages regardless of their payload size
-	 *            should be stored in Amazon S3. Default: false
-	 */
-	public void setAlwaysThroughS3(boolean alwaysThroughS3) {
-		this.alwaysThroughS3 = alwaysThroughS3;
-	}
-
-	/**
-	 * Sets whether or not all messages regardless of their payload size should
-	 * be stored in Amazon S3.
-	 *
-	 * @param alwaysThroughS3
-	 *            Whether or not all messages regardless of their payload size
-	 *            should be stored in Amazon S3. Default: false
-	 * @return the updated ExtendedClientConfiguration object.
-	 */
-	public ExtendedClientConfiguration withAlwaysThroughS3(boolean alwaysThroughS3) {
-		setAlwaysThroughS3(alwaysThroughS3);
-		return this;
-	}
-
-	/**
-	 * Checks whether or not all messages regardless of their payload size are
-	 * being stored in Amazon S3.
-	 *
-	 * @return True if all messages regardless of their payload size are being
-	 *         stored in Amazon S3. Default: false
-	 */
-	public boolean isAlwaysThroughS3() {
-		return alwaysThroughS3;
-	}
-}
+public class ExtendedClientConfiguration extends PayloadStorageConfiguration {
+
+    private boolean cleanupS3Payload = true;
+    private boolean useLegacyReservedAttributeName = true;
+
+    public ExtendedClientConfiguration() {
+        super();
+        this.setPayloadSizeThreshold(SQSExtendedClientConstants.DEFAULT_MESSAGE_SIZE_THRESHOLD);
```

Comment:
could you please explain why we need this line?

Reply:
The common lib PayloadStorageConfiguration by default sets `payloadSizeThreshold=0` so change it to default to 262144 when creating a ExtendedClientConfiguration